rbac_api_with_domains_test.go

```diff
@@ -267,3 +268,12 @@ func TestGetAllDomains(t *testing.T) {
 
 	testGetAllDomains(t, e, []string{"domain1", "domain2"})
 }
+
```

Comment:
I think singular test case is not sufficient for such a complex feature. At least one which supports deep hierarchy should also be done. Like here:

https://github.com/casbin/casbin/issues/718#issuecomment-889785940

Reply:
More specifically: Multiple subjects, multiple objects, multiple domains

1. Test cases for accessing data in an upper domain from a lower domain (Failure)
2. Test cases for accessing data in a lower domain from a middle domain (Success)
3. Test cases for accessing data in a sibling domain from another sibling (Failure)
4. Test cases for accessing data in a sibling domain from parent (success)

Reply:
Added more tests

Reply:
I tried to implement this in https://github.com/casbin/node-casbin/pull/380

I think it's not possible to define `g2` in the format of `child, parent`. At least in nodeJS, it provided incorrect results, but maybe `golang` has a different RoleManager definition.

You can try to replicate the test case I did for nodeJS. I think it provides quite good coverage on all the possible use cases. Especially the 4 domain structure is essential.

I have tried to comment on why certain decisions were made in the pull request and a little bit about the thought behind the hierarchy model.